### ISSUE TITLE
Migrate system replicas to new cc sizes

### ISSUE THREAD

**Issue by antiguru** [MaterializeInc/materialize#24983]
## Problem

Currently, our system replicas use legacy cluster sizes. The system replicas are part of the catalog, so we need a migration to switch them over to the new sizes.

This is complicated because we need to use the new sizes everywhere, in cloud and tests.

## Solution

* Implement a for system replicas to the new cc sizes, but only enable it once the cc sizes are in GA.

## See also

* Related: https://github.com/MaterializeInc/cloud/issues/8390

**Comment by benesch** [comment]
We should also consider sizing down the `mz_system` cluster. In all of production, max CPU usage is about 20% and max memory usage is about 10%.

<img width="827" alt="image" src="https://github.com/MaterializeInc/materialize/assets/882976/cd1cc99a-8f38-4f54-8e9a-f4c380b55ed7">

<img width="846" alt="image" src="https://github.com/MaterializeInc/materialize/assets/882976/3d1ff39c-8544-4cfe-ab08-23dd5d57e48b">

Currently `mz_system` is a `2xsmall`, so a `50cc`. We should be able to scale this down to a `25cc` no problem.

**Comment by benesch** [comment]
Then again, a 10cc is a really underpowered cluster for debugging queries. Maybe 25cc is the way to thread the needle for now. One day we'll have automatic cluster suspend/resume (https://github.com/MaterializeInc/materialize/issues/23132) and autoscaling (https://github.com/MaterializeInc/materialize/issues/22568), and none of this will matter. `mz_system` will only turn on when the usage exporter is running or when someone runs debugging queries, and it will automatically scale up to the right size. But until we have those features, a 25cc seems like the way to go.

**Comment by jubrad** [comment]
If a replica for mz_introspection or mz_system clusters had been sized up for some reason should they still be moved to the standard sizes we're suggesting here or should the migration look for these outliers and we can handle them case by case by creating a new replica of the correct size and dropping the old replica. (arguably we could migrate all clusters this way, but I don't think that's the route we want to take).

**Comment by benesch** [comment]
> If a replica for mz_introspection or mz_system clusters had been sized up for some reason should they still be moved to the standard sizes we're suggesting here or should the migration look for these outliers and we can handle them case by case by creating a new replica of the correct size and dropping the old replica.

I'd say handle them case by case, yeah.

> (arguably we could migrate all clusters this way, but I don't think that's the route we want to take).

Wait, honestly ... maybe that's exactly what we should do? It's a lot less kind of scary code. It's just a simple script that we'd run.

That might work for the migration to managed clusters, too.

There'd still be work to do to change the defaults for new environments in the code. But I kind of love the idea of just doing this manually for the migration.

**Comment by benesch** [comment]
Yeah, I think that's fine that we don't have an automatic test! It's a one-off script, so fine to do a one-off test. Let's write the script and run it against the production sandbox and production analytics accounts. If it works there, it'll work everywhere, and we can ask the release manager to do the rest during the following release window.
